test/gtest/ucp/test_ucp_mmap.cc

```diff
@@ -355,18 +356,28 @@ bool test_ucp_mmap::disable_proto() const
     return get_variant_value() == VARIANT_PROTO_DISABLE;
 }
 
-void test_ucp_mmap::expect_same_distance(const ucs_sys_dev_distance_t &dist1,
-                                         const ucs_sys_dev_distance_t &dist2)
+void test_ucp_mmap::check_distance_precision(double rkey_value,
+                                             double topo_value,
+                                             size_t pack_min,
+                                             size_t pack_max)
 {
     /* Expect the implementation to always provide a reasonable precision w.r.t.
      * real-world bandwidth and latency ballpark numbers.
      */
-    EXPECT_NEAR(dist1.bandwidth, dist2.bandwidth, 600e6); /* 600 MBs accuracy */
-    EXPECT_NEAR(dist1.latency, dist2.latency, 20e-9); /* 20 nsec accuracy */
+    double allowed_diff_ratio = 1 - UCS_FP8_PRECISION;
+
+    /* Even if topo_value crosses the pack borders, it still can be not
+       capped in case of successfull cache entry catch */
+    if (((topo_value <= pack_min) && (rkey_value != pack_min)) ||
+        ((topo_value >= pack_max) && (rkey_value != pack_max))) {
+        EXPECT_NEAR(rkey_value, topo_value, topo_value * allowed_diff_ratio);
+    }
```

Comment:
why are only out of range values checked now?

Reply:
Good catch, thanks, fixed!

Reply:
Good catch, thanks! Fixed.